### ISSUE TITLE
Subresource Path Is Displayed Incorrectly

### ISSUE THREAD

**Issue by yongk802** [swagger-api/swagger-core#2632]
| Q                                 | A
| --------------------------------- | -------
| Bug or feature request?           |  Bug
| Which Swagger-Core version?       | 1.5.17
| Which Java version?               | 1.8.0_144
| Which JAX-RS framework & version? | Jackson 2.9.1
 
Subresource paths are not displayed correctly if the subresource has a path on it at the class level.

I ran into this because I wanted to override a Path on a Jersey resource from another project. I've worked around that, but this still seems worth reporting. 

### Expected Behavior:
When pointing to a specific subresource from a parent resource, giving it a Path, the Path on the subresource at the class level should be ignored by Swagger, e.g., /v1/parent.

### Actual Behavior: 
It is not ignored and is concatenated together with the parent path - also with no slash, .e.g.,
/v1/parentsubresource.

Jersey correctly registers the resource and accepts requests to the endpoint (/v1/parent), but the generated Swagger documentation is incorrect. There is no such resource at /v1/parentsubresource or /v1/parent/subresource, nor should there be.

Here is a simple runnable project created from a dropwizard archetpye where this can be seen:

```
https://github.com/ef-labs/swagger-subresource-issue
```

**Comment by FuweiChin** [comment]
Same issue in swagger-jaxrs2 2.0.0-rc4

Java documentation says that `javax.ws.rs.Path`

> Identifies the URI path that a resource class or class method will serve requests for. 
> 
> Paths are relative. For an annotated class the base URI is the application path, see ApplicationPath. For an annotated method the base URI is the effective URI of the containing class. For the purposes of absolutizing a path against the base URI , **a leading '/' in a path is ignored** and base URIs are treated as if they ended in '/'. e.g.:

```java
@Path("widgets")
public class WidgetsResource {
  @GET
  String getList() {...}

  @GET @Path("{id}")
  String getWidget(@PathParam("id") String id) {...}
}
```

so I followed the example but swagger doesn't support it.

**Comment by frantuma** [comment]
fixed in #2697 (master) and #2700 (2.0)
